luminoso_api/client.py

```diff
@@ -487,14 +489,34 @@ def get_token_filename():
                         '.luminoso',
                         'tokens.json')
 
-def get_root_url(url):
+def get_root_url(url, warn=True):
     """
-    If we have to guess a root URL, assume it contains the scheme,
-    hostname, and one path component, as in "https://api.luminoso.com/v4".
+    Guessing the root URL is slightly complicated, because the API may be at
+    /v4 or at /api/v4, depending on whether you're accessing the API through
+    its old URL or its new one.
     """
-    # make sure it's a complete URL, not a relative one
-    assert ':' in url
-    return '/'.join(url.split('/')[:4])
+    parsed_url = urlparse(url)
+
+    # Make sure it's a complete URL, not a relative one.
+    if not parsed_url.scheme:
+        raise ValueError('Please supply a full URL, beginning with http:// '
+                         'or https:// .')
+
+    # If the path starts with v4, use that for the base URL, but issue a
+    # warning about the URL moving.
+    if parsed_url.path.startswith('/v4'):
+        if warn:
+            logger.warning('While the old URL will continue to work, please '
+                           'be aware that the Luminoso API has moved to %s',
+                           URL_BASE)
```

Comment:
It applies to any API running via nginx (so it should also use a URL that will work if you connect to our master-staging API).

In general, though, you're right: the URL it gives will be the deployed API, and if you try to connect to your own computer at /v4, but your computer doesn't have a redirect set up (because it's not running the API through nginx), then yes, things will fail for you.  I am, however, fundamentally OK with this; I don't think it's necessary to go through any convoluted steps to make sure that *we* get the right error messages.  Warnings and redirects that only apply to external users seem fine to me.